### PR TITLE
Persistent Character location

### PR DESCRIPTION
**Pull request by ahoiahoi**
Store player location in the DB

### CODE REVIEW

**Comment by Rawaho** [comment]
Location and world is already stored on every grid entity, there is no need to have them duplicated on the player too.
My suggestion would be using `OnRelocate` override to set the save mask for location and when the actual save occurs pull the location from the grid entity.

**Comment by ahoiahoi** [comment]
@Rawaho good idea, thank you

File: Source/NexusForever.WorldServer/Game/Entity/Player.cs [hunk 1]
```diff
@@ -145,6 +149,15 @@ public override void OnAddToMap(BaseMap map, uint guid, Vector3 vector)
             Session.EnqueueMessageEncrypted(new ServerPlayerEnteredWorld());
 
             IsLoading = false;
+
+            Position = vector;
+            Map.Entry.Id = (ushort)map.Entry.Id;
```
Comment (Rawaho) [inline Source/NexusForever.WorldServer/Game/Entity/Player.cs]: Both position and map are set before this in `base.OnAddToMap`, no need to do it again

**Review by Rawaho — CHANGES_REQUESTED** [review]
Just a single small change

**Comment by ahoiahoi** [comment]
@Rawaho updated

**Review by Rawaho — APPROVED** [review]

**Pull request merged**